src/wallet/wallet2.cpp

```diff
@@ -7802,6 +7840,15 @@ std::vector<wallet2::pending_tx> wallet2::create_transactions_single(const crypt
 
 std::vector<wallet2::pending_tx> wallet2::create_transactions_from(const cryptonote::account_public_address &address, bool is_subaddress, std::vector<size_t> unused_transfers_indices, std::vector<size_t> unused_dust_indices, const size_t fake_outs_count, const uint64_t unlock_time, uint32_t priority, const std::vector<uint8_t>& extra, bool trusted_daemon)
 {
+  //ensure device is let in NONE mode in any case
+  hw::device &hwdev = m_account.get_device();
+  std::unique_lock<hw::device> hwdev_lock (hwdev);
+  struct reset_hwdev {
+    hw::device& ref;
+    reset_hwdev(hw::device& dev) : ref(dev) { }
+    ~reset_hwdev() { ref.set_mode(hw::device::NONE);}
+  } rst(hwdev);
```

Comment:
That class can be factored, it's used three times so far.

Reply:
ok